Sources/BrowserServicesKit/SecureVault/SecureVaultModels.swift

```diff
@@ -500,18 +502,18 @@ extension Array where Element == SecureVaultModels.WebsiteAccount {
 
     // Last Updated > Alphabetical Domain > Alphabetical Username > Empty Usernames
     private func compareAccount(_ account1: SecureVaultModels.WebsiteAccount, _ account2: SecureVaultModels.WebsiteAccount) -> Bool {
-        if !account1.username.isEmpty && account2.username.isEmpty {
+        if !(account1.username ?? "").isEmpty && (account2.username ?? "").isEmpty {
             return true
-        } else if account1.username.isEmpty && !account2.username.isEmpty {
+        } else if (account1.username ?? "").isEmpty && !(account2.username ?? "").isEmpty {
             return false
         } else if account1.lastUpdated.withoutTime != account2.lastUpdated.withoutTime {
             return account1.lastUpdated.withoutTime > account2.lastUpdated.withoutTime
-        } else if account1.domain != account2.domain {
-            return account1.domain < account2.domain
-        } else if !account1.username.isEmpty && !account2.username.isEmpty {
-            return account1.username < account2.username
+        } else if let domain1 = account1.domain, let domain2 = account2.domain, domain1 != domain2 {
+            return domain1 < domain2
+        } else if let username1 = account1.username, let username2 = account2.username, !username1.isEmpty && !username2.isEmpty {
+            return username1 < username2
```

Comment:
I wonder, how we should compare empty domain/username with non empty one? If one is nil it will skip the comparison now, which I'm not sure is what we want.

Reply:
Let me update this function and treat nils as empty strings.

Reply:
Done, it should be simpler to understand now. I didn't change the logic vs the initial implementation (before this PR), and only applied coalescing nil domains to empty strings.